WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -603,9 +607,13 @@ private extension MainTabBarController {
             guard let self = self else { return }
             Task { @MainActor in
                 await self.showProductDetails(for: error)
+                self.analytics.track(event: .ImageUpload
+                    .failureSavingProductAfterImageUploadNoticeTapped(productOrVariation: error.productOrVariationEventProperty))
             }
         })
         noticePresenter.enqueue(notice: notice)
+        analytics.track(event: .ImageUpload
+            .failureSavingProductAfterImageUploadNoticeShown(productOrVariation: error.productOrVariationEventProperty))
```

Comment:
Great catch on this, Sharma mentioned this notice presenter behavior in a recent separate PR as well but I forgot about this 😓 In https://github.com/woocommerce/woocommerce-ios/pull/7196/commits/33b4dd84bffa6a6c901e3959caa3768dc3f62175, I updated the `enqueue` function to return a discardable boolean value that indicates whether the notice **can be displayed**. Though, enqueuing a notice successfully doesn't mean that it's being displayed right away and I don't see an easy way to return a value when a given notice is being displayed without changing the interface and implementation too much (will have to mark the function as async and have some way to associate each notice with a closure or publisher). I think this is fine for our use case since an enqueued notice is most likely displayed at some point unless the user terminates the app. Any suggestions are welcome!

Reply:
Yes, I agree that the current solution is good enough for now 👍